src/cpp/shared/game/event_loop.cpp

```diff
@@ -20,3 +20,24 @@ void __fastcall Process1WindowsMessage() {
 }
 
 
+unsigned long UTickCount() {
+  return KBTickCount();
+}
+
+/*
+ * This is a slightly hacky fix to the bug described in https://www.celestialheavens.com/forum/7/16792 ,
+ * where a signed overflow of GetTickCount() can cause the game to wait for up to 25 days
```

Comment:
Do you want to add an upper bound to prevent the game ever hanging forever again?  Is it ever reasonable for `DelayTilMilli` to wait for more than a few seconds?

Reply:
There were two problems that combined to form these long delays:

The first is the possibility of wraparound.

The second is that some code accidentally put in a very small number into DelayTillMilli.

The end result is waiting a very long time.

With this change, the only way DelayTillMilli could wait a very long time is if the caller passed in a very large number. In which case, either they really did want a very long wait, or it was a mistake, in which case we shouldn't hide it (though I guess we can detect it and exit with an error message). In any case, this change cannot introduce any new very-long-hangs where there were none before.